checks/allprocesses_windows.go

```diff
@@ -0,0 +1,306 @@
+		UserModeTime          uint64
+		WorkingSetSize        uint64
+	*/
+}
+
+type IO_COUNTERS struct {
+	ReadOperationCount  uint64
+	WriteOperationCount uint64
+	OtherOperationCount uint64
+	ReadTransferCount   uint64
+	WriteTransferCount  uint64
+	OtherTransferCount  uint64
+}
+
+func getProcessMemoryInfo(h syscall.Handle, mem *process.PROCESS_MEMORY_COUNTERS) (err error) {
+	r1, _, e1 := syscall.Syscall(procGetProcessMemoryInfo.Addr(), 3, uintptr(h), uintptr(unsafe.Pointer(mem)), uintptr(unsafe.Sizeof(*mem)))
+	if r1 == 0 {
+		if e1 != 0 {
+			err = error(e1)
+		} else {
+			err = syscall.EINVAL
+		}
+	}
+	return
+}
+func getProcessHandleCount(h syscall.Handle, count *uint32) (err error) {
+	r1, _, e1 := procGetProcessHandleCount.Call(uintptr(h), uintptr(unsafe.Pointer(count)))
+	if r1 == 0 {
+		return e1
+	}
+	return nil
+}
+
+func getProcessIoCounters(h syscall.Handle, counters *IO_COUNTERS) (err error) {
+	r1, _, e1 := procGetProcessIoCounters.Call(uintptr(h), uintptr(unsafe.Pointer(counters)))
+	if r1 == 0 {
+		return e1
+	}
+	return nil
+}
+
+func getProcessMapFromWmi() (map[uint32]Win32_Process, error) {
+	var dst []Win32_Process
+	q := wmi.CreateQuery(&dst, "")
+	err := wmi.Query(q, &dst)
+	if err != nil {
+		return map[uint32]Win32_Process{}, err
+	}
+	if len(dst) == 0 {
+		return map[uint32]Win32_Process{}, fmt.Errorf("could not get Process")
+	}
+	results := make(map[uint32]Win32_Process)
+	for _, proc := range dst {
+		results[proc.ProcessID] = proc
+	}
+	return results, nil
+
+}
+func getAllProcesses() (map[int32]*process.FilledProcess, error) {
+	allProcsSnap := w32.CreateToolhelp32Snapshot(w32.TH32CS_SNAPPROCESS, 0)
+	if allProcsSnap == 0 {
+		return nil, syscall.GetLastError()
+	}
+	procs := make(map[int32]*process.FilledProcess)
+
+	defer w32.CloseHandle(allProcsSnap)
+	var pe32 w32.PROCESSENTRY32
+	pe32.DwSize = uint32(unsafe.Sizeof(pe32))
+
+	fromWmi, _ := getProcessMapFromWmi()
+
+	for success := w32.Process32First(allProcsSnap, &pe32); success; success = w32.Process32Next(allProcsSnap, &pe32) {
+		pid := pe32.Th32ProcessID
+		ppid := pe32.Th32ParentProcessID
+
+		// 0x1000 is PROCESS_QUERY_LIMITED_INFORMATION, but that constant isn't
+		// defined in syscall
+		procHandle, err := syscall.OpenProcess(0x1000, false, uint32(pid))
+		if err != nil {
+			continue
+		}
+		defer syscall.CloseHandle(procHandle)
+
+		var CPU syscall.Rusage
+		if err = syscall.GetProcessTimes(procHandle, &CPU.CreationTime, &CPU.ExitTime, &CPU.KernelTime, &CPU.UserTime); err != nil {
+			continue
+		}
+
+		var handleCount uint32
+		if err = getProcessHandleCount(procHandle, &handleCount); err != nil {
+			continue
+		}
+
+		var pmemcounter process.PROCESS_MEMORY_COUNTERS
+		if err = getProcessMemoryInfo(procHandle, &pmemcounter); err != nil {
+			continue
+		}
+
+		// shell out to getprocessiocounters for io stats
+		var ioCounters IO_COUNTERS
+		if err = getProcessIoCounters(procHandle, &ioCounters); err != nil {
+			continue
```

Comment:
for all these `continue` cases I'm in favor of having some debug or trace-level logging. We went without this in the initial cut of the linux AllProcesses and it bit us when we had an issue in the wild where processes were skipped without any logging